pywbemtools/pywbemcli/_pywbemcli_operations.py

```diff
@@ -288,41 +290,240 @@ def PyWbemcliQueryInstances(self, FilterQueryLanguage, FilterQuery,
+          The mock scripts can be used for any kind of setup of the mock
+          environment, for example for creating namespaces, for defining
+          provider classes and registering providers, or for adding CIM objects
+          either directly through add_cimobjects() or by compiling MOF files.
+
+          Mock scripts support two approaches for passing the connection and
+          server objects they should operate on:
+
+          * via a setup() function defined in the mock script. This is the
+            recommended approach, and it supports caching. The setup()
+            function has the following parameters:
+
+              conn (pywbem_mock.FakedWBEMConnection): The mock connection.
+
+              server (pywbem.WBEMServer): The server object for the mock
+                connection.
+
+              verbose (bool): Verbose flag from the command line.
+
+          * via global variables made available to the mock script. This
+            approach prevents caching. The following global variables are
+            made available:
+
+              CONN (pywbem_mock.FakedWBEMConnection): The mock connection.
+
+              SERVER (pywbem.WBEMServer): The server object for the mock
+                connection.
+
+              VERBOSE (bool): Verbose flag from the command line.
+
+        Parameters:
+
+          self (pywbem_mock.FakedWBEMConnection): The mock connection.
+
+          server (pywbem.WBEMServer): The server object for the mock connection.
+
+          file_path_list (list of string): The path names of the input files
+            for building the mock environment, from the connection definition.
+
+          connections_file (string): Path name of the connections file.
+
+          connection_name (string): The name of the connection definition in
+            the connections file.
+
+          verbose (bool): Verbose flag from the command line.
+
+        Raises:
+          MockFileError: Mock file does not exist.
+          MockMOFCompileError: Mock MOF file fails to compile.
+          MockScriptError: Mock script fails to execute.
+          SetupNotSupportedError (py<3.5): New-style setup in mock script not
+            supported.
         """
+
+        # Check that the input files exist. Since we loop through them multiple
+        # times, we check that once.
         for file_path in file_path_list:
             if not os.path.exists(file_path):
-                raise IOError('No such file: {}'.format(file_path))
+                raise mockscripts.MockFileError(
+                    "Mock file does not exist: {}".format(file_path))
+
+        # The connections file is set if a named connection is used, i.e.
+        # when specifying the -n general option. It is unset when the -s or -m
+        # general options were specified. In the latter case, no caching happens
+        # because there is no connection definition context under which that
```

Comment:
Suggest change "under which that could be done" to "which is required for caching"

I am trying to get close to clean statement that "caching is only executed for named connections"

Reply:
DONE